lib/Echidna/Solidity.hs

```diff
@@ -114,23 +138,27 @@ loadSpecified name cs = let ensure l e = if l == mempty then throwM e else pure
   c <- choose cs name
   q <- view (hasLens . quiet)
   liftIO $ do
-    when (isNothing name && length cs > 1) $
+    when (isNothing name && length cs > 1 && not q) $
       putStrLn "Multiple contracts found in file, only analyzing the first"
     unless q . putStrLn $ "Analyzing contract: " <> unpack (c ^. contractName)
 
   -- Local variables
-  (SolConf ca d ads b pref _ _) <- view hasLens
+  (SolConf ca d ads b pref _ libs _) <- view hasLens
   let bc = c ^. creationCode
       blank = populateAddresses (ads |> d) b (vmForEthrunCreation bc)
       abi = liftM2 (,) (view methodName) (fmap snd . view methodInputs) <$> toList (c ^. abiMap)
       (tests, funs) = partition (isPrefixOf pref . fst) abi
 
+  -- Select libraries
+  ls <- mapM (choose cs . Just . pack) libs
+
   -- Make sure everything is ready to use, then ship it
   mapM_ (uncurry ensure) [(abi, NoFuncs), (tests, NoTests), (funs, OnlyTests)] -- ABI checks
   ensure bc (NoBytecode $ c ^. contractName)                                   -- Bytecode check
   case find (not . null . snd) tests of
     Just (t,_) -> throwM $ TestArgsFound t                                     -- Test args check
-    Nothing    -> (, funs, fst <$> tests) <$> execStateT (execTx $ Tx (Right bc) d ca 0) blank
+    Nothing    -> do vm <- loadLibraries ls addrLibrary d blank
```

Comment:
we can just `loadLibraries [...] >>= (, funs, fst <$> tests) [...]` and not need do notation here

Reply:
Done. The `\vm -> .. vm` seems to be necessary to compile (maybe because the pair syntax?). Feel free to change it, if you find a way to do it.